From 52a4bf740d52cd48660dd6e6ef171801fd0f50aa Mon Sep 17 00:00:00 2001
From: Tim Murray <timmurray@google.com>
Date: Tue, 19 Jan 2016 16:35:17 -0800
Subject: [PATCH] dm-verity: run in a WQ_HIGHPRI workqueue

Running dm-verity in a standard workqueue results in IO competing for
CPU time with standard user apps, which can lead to pipeline bubbles and
seriously degraded performance. Move to a WQ_HIGHPRI workqueue to
protect against that.

bug 25392275

Change-Id: Ic65d7bd6f04e4d77780119e926a50e71323575f0
---
 drivers/md/dm-verity.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/md/dm-verity.c b/drivers/md/dm-verity.c
index ac099847a4b..26e9f05b15d 100644
--- a/drivers/md/dm-verity.c
+++ b/drivers/md/dm-verity.c
@@ -855,7 +855,7 @@ static int verity_ctr(struct dm_target *ti, unsigned argc, char **argv)
 	}
 
 	/* WQ_UNBOUND greatly improves performance when running on ramdisk */
-	v->verify_wq = alloc_workqueue("kverityd", WQ_CPU_INTENSIVE | WQ_MEM_RECLAIM | WQ_UNBOUND, num_online_cpus());
+	v->verify_wq = alloc_workqueue("kverityd", WQ_HIGHPRI | WQ_MEM_RECLAIM | WQ_UNBOUND, num_online_cpus());
 	if (!v->verify_wq) {
 		ti->error = "Cannot allocate workqueue";
 		r = -ENOMEM;
